### PR TITLE
Documentation & working example for Authelia OAuth SSO

### PR DESCRIPTION
**Pull request by DennisGaida**
### Component/Part
Documentation

### Description
This PR adds documentation and a working example for setting up OAuth SSO with Authelia ([www.authelia.com](https://github.com/authelia/authelia))

### Steps

- [x] Added documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/content/guides/auth/authelia.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# OAuth with Authelia SSO (self-hosted)
+
+## Prerequisites
+
+This guide assumes you have run and configured [Authelia](https://www.authelia.com/). If you want to get Authelia running quickly, there are example docker-compose files in the Authelia Github repository: https://github.com/authelia/authelia/tree/master/examples/compose.
```
Comment (DerMolly) [inline docs/content/guides/auth/authelia.md:5]:
```suggestion
This guide assumes you have run and configured [Authelia](https://www.authelia.com/). If you want to get Authelia running quickly, there are example docker-compose files in the [Authelia Github repository](https://github.com/authelia/authelia/tree/master/examples/compose).
```

File: docs/content/guides/auth/authelia.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
```
Comment (DerMolly) [inline docs/content/guides/auth/authelia.md:6]:
```suggestion
Also this guides assumes you run HedgeDoc via a [Docker container](../../setup/docker.md). Find out how the mentioned config environment variables are mapped to entries in the config file at our [configuration page](../../configuration.md).

```

File: docs/content/guides/auth/authelia.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  oidc:
+    hmac_secret: <hmac secret here> # use docker secrets for this
+    issuer_private_key: <issuer private key secret here> # use docker secrets for this
+    access_token_lifespan: 1h
+    authorize_code_lifespan: 1m
+    id_token_lifespan: 1h
+    refresh_token_lifespan: 90m
+    enable_client_debug_messages: false
+    clients:
+      - id: HedgeDoc
```
Comment (DerMolly) [inline docs/content/guides/auth/authelia.md:24]:
```suggestion
      - id: HedgeDoc # this should be changed to something more secure
```

**Review by DerMolly — CHANGES_REQUESTED** [review]
Hey @DennisGaida,

thank you for this contribution. I added some suggestion what could be improved with the guide.

Somebody will also need to verify this works by trying out this guide, as I did not do that yet.

Also please make sure that you [Sign-Off your commits](https://docs.pi-hole.net/guides/github/how-to-signoff/). Note that signing off your commits accepts our DCO, a legally binding statement. Please carefully read our [contribution docs](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md#certificate-of-origin).

**Comment by DennisGaida** [comment]
Hmhmm I guess I should have first accept the changes, then `signoff` the previous commits and force push?

**Comment by DennisGaida** [comment]
Great, now I can't sign off on these three review/change commits I just clicked in the GH UI since it doesn't automatically sign off. Of course I can't force push to this repo. How do we proceed - I don't think I can sign-off on these changes?

**Comment by mrdrogdrog** [comment]
You can force push into your own fork.

**Review by davidmehren — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Pull request merged**